datafusion/optimizer/src/decorrelate.rs

```diff
@@ -148,7 +148,7 @@ impl TreeNodeRewriter for PullUpCorrelatedExpr {
                         let mut plan =
                             LogicalPlanBuilder::from((*plan_filter.input).clone());
                         if let Some(expr) = conjunction(subquery_filters) {
-                            plan = plan.filter(expr)?
+                            plan = plan.filter(expr, 20)?
```

Comment:
This seems a case where the optimizer doesn't really optimize an existing filter, but add a filter, so I think we can only default it